src/Playwright.Tests/BrowserTypeConnectTests.cs

```diff
@@ -80,71 +82,396 @@ public void TearDown()
+        [Skip(SkipAttribute.Targets.Windows)]
+        public async Task ShouldTimeoutInConnectWhileConnecting()
+        {
+            var exception = await PlaywrightAssert.ThrowsAsync<PlaywrightException>(async () => await BrowserType.ConnectAsync(_browserServer.WSEndpoint, new BrowserTypeConnectOptions { Timeout = 1 }));
+            StringAssert.Contains("Timeout 1ms exceeded", exception.Message);
+        }
+
+        [PlaywrightTest("browsertype-connect.spec.ts", "should support slowmo option")]
+        public async Task ShouldSupportSlowMo()
         {
+            var browser = await BrowserType.ConnectAsync(_browserServer.WSEndpoint, new BrowserTypeConnectOptions { SlowMo = 200 });
+            var context = await browser.NewContextAsync();
+            var time1 = DateTime.Now;
+            var page = await context.NewPageAsync();
+            await page.GotoAsync(Server.EmptyPage);
+            var time2 = DateTime.Now;
+            Assert.AreEqual(Server.EmptyPage, page.Url);
+            Assert.Greater((time2 - time1).TotalMilliseconds, 200);
         }
 
         [PlaywrightTest("browsertype-connect.spec.ts", "disconnected event should be emitted when browser is closed or server is closed")]
-        [Ignore("SKIP WIRE")]
-        public void DisconnectedEventShouldBeEmittedWhenBrowserIsClosedOrServerIsClosed()
+        public async Task DisconnectedEventShouldBeEmittedWhenBrowserIsClosedOrServerIsClosed()
         {
+            var browserOne = await BrowserType.ConnectAsync(_browserServer.WSEndpoint);
+            var pageOne = browserOne.NewPageAsync();
+
+            var browserTwo = await BrowserType.ConnectAsync(_browserServer.WSEndpoint);
+            var pageTwo = browserTwo.NewPageAsync();
+
+            int browserOneCloseCount = 0;
+            int browserTwoCloseCount = 0;
+            browserOne.Disconnected += (_, e) => browserOneCloseCount++;
+            browserTwo.Disconnected += (_, e) => browserTwoCloseCount++;
+
+            await browserOne.CloseAsync();
+            Assert.AreEqual(browserOneCloseCount, 1);
+            Assert.AreEqual(browserTwoCloseCount, 0);
+
+            await browserTwo.CloseAsync();
+            Assert.AreEqual(browserOneCloseCount, 1);
+            Assert.AreEqual(browserTwoCloseCount, 1);
         }
 
-        [PlaywrightTest("browsertype-connect.spec.ts", "should handle exceptions during connect")]
-        [Ignore("SKIP WIRE")]
-        public void ShouldHandleExceptionsDuringConnect()
+        [PlaywrightTest("browsertype-connect.spec.ts", "disconnected event should have browser as argument")]
+        public async Task DisconnectedEventShouldHaveBrowserAsArguments()
         {
+            var browser = await BrowserType.ConnectAsync(_browserServer.WSEndpoint);
+            IBrowser disconneced = null;
+            browser.Disconnected += (_, e) => disconneced = e;
+            await browser.CloseAsync();
+            Assert.AreEqual(browser, disconneced);
         }
 
```

Comment:
"should handle exceptions during connect" test is missing

Reply:
This seems to be an implementational detail for NodeJS. Test isn't present on playwright java as well - https://github.com/microsoft/playwright-java/blob/main/playwright/src/test/java/com/microsoft/playwright/TestBrowserTypeConnect.java#L211